va/validation-authority.go

```diff
@@ -487,6 +487,11 @@ func (va *ValidationAuthorityImpl) validateDNS01(identifier core.AcmeIdentifier,
 	for _, element := range txts {
 		if subtle.ConstantTimeCompare([]byte(element), []byte(authorizedKeysDigest)) == 1 {
 			challenge.Status = core.StatusValid
+			challenge.ValidationRecord = []core.ValidationRecord{
+				core.ValidationRecord{
+					Hostname: identifier.Value,
```

Comment:
This is the change I mention doing in another change in description since it will require a bunch of changes to the `bdns` package which is somewhat out of the scope of this PR.